### PR TITLE
strongbox integration

### PR DESCRIPTION
**Pull request by alkar**
Add the `strongboxKeyringSecretRef` attribute to the Application CRD.
Use the `.strongbox_keyring` in the Secret when cloning the repo to apply a namespace.
Remove strongbox `initContainer` from the base, the binary and setup is now baked in the image.

### CODE REVIEW

File: apis/kubeapplier/v1alpha1/application_types.go [hunk 1]
```diff
@@ -41,6 +41,12 @@ type ApplicationSpec struct {
 	// +optional
 	// +kubebuilder:default=false
 	ServerSideApply bool `json:"serverSideApply,omitempty"`
+
+	// StrongboxKeyringSecretRef reference a Secret in the same namespace as the
+	// Application that contains a single item, named '.strongbox_keyring' with
+	// any strongbox keys required to decrypt the files before applying.
+	// +optional
+	StrongboxKeyringSecretRef string `json:"StrongboxKeyringSecretRef,omitempty"`
```
Comment (ribbybibby) [inline apis/kubeapplier/v1alpha1/application_types.go:49]:
```suggestion
	StrongboxKeyringSecretRef string `json:"strongboxKeyringSecretRef,omitempty"`
```

File: manifests/base/crd/kube-applier.io_applications.yaml [hunk 2]
```diff
@@ -41,6 +41,12 @@ spec:
           spec:
             description: ApplicationSpec defines the desired state of Application
             properties:
+              StrongboxKeyringSecretRef:
```
Comment (ribbybibby) [inline manifests/base/crd/kube-applier.io_applications.yaml:44]:
```suggestion
              strongboxKeyringSecretRef:
```

**Review by ribbybibby — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -112,8 +112,29 @@ spec:
...
+If the Application spec contains a `strongboxKeyringSecretRef` value, the value
+should be the name of a Secret resource (in the namespace where the Application
+resides), which contains a key named `.strongbox_keyring` with its value being
+a valid strongbox keyring file. That keyring is subsequently used when applying
+the Application, allowing for decryption of files under the `repositoryPath`.
+
+The secret containing the strongbox keyring should itself be version controlled
+to prevent kube-applier from pruning it. However, since it is a secret itself
+and would need be encrypted as well in git, it must be created manually the
+first time (or after any changes to its contents).
```
Comment (george-angel) [inline README.md:136]: it's (I think, @ribbybibby might need to correct me)

**Review by george-angel — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (alkar) [reply README.md:136]: Oops, didn't refresh before merging, sorry!
But I believe it is "its" (possessive adjective)

**Pull request merged**